tgui/packages/tgui/interfaces/PowerMonitor/Apc.tsx

```diff
@@ -81,48 +83,112 @@ export const PowerMonitorApcGlobal = (_props, context) => {
+          Area
+        </Header>
+      </Table.Cell>
       <Tooltip content="Equipment">
         <Table.Cell header collapsing>
-          Eqp.
+          <Header
+            sortDirection={props.state?.field === ApcTableHeaderColumns.Equipment ? props.state.dir : null}
+            onSortClick={() => props.setSortBy(ApcTableHeaderColumns.Equipment)}>
+            Eqp.
+          </Header>
         </Table.Cell>
       </Tooltip>
       <Tooltip content="Lighting">
         <Table.Cell header collapsing>
-          Lgt.
+          <Header
+            sortDirection={props.state?.field === ApcTableHeaderColumns.Lighting ? props.state.dir : null}
+            onSortClick={() => props.setSortBy(ApcTableHeaderColumns.Lighting)}>
+            Lgt.
+          </Header>
         </Table.Cell>
       </Tooltip>
       <Tooltip content="Environment">
         <Table.Cell header collapsing>
-          Env.
+          <Header
+            sortDirection={props.state?.field === ApcTableHeaderColumns.Environment ? props.state.dir : null}
+            onSortClick={() => props.setSortBy(ApcTableHeaderColumns.Environment)}>
+            Env.
+          </Header>
         </Table.Cell>
       </Tooltip>
       <Table.Cell header textAlign="right">
-        Load
+        <Header
+          sortDirection={props.state?.field === ApcTableHeaderColumns.Load ? props.state.dir : null}
+          onSortClick={() => props.setSortBy(ApcTableHeaderColumns.Load)}>
+          Load
+        </Header>
       </Table.Cell>
       <Table.Cell header textAlign="right">
-        Cell Charge
+        <Header
+          sortDirection={props.state?.field === ApcTableHeaderColumns.CellCharge ? props.state.dir : null}
+          onSortClick={() => props.setSortBy(ApcTableHeaderColumns.CellCharge)}>
+          Cell Charge
+        </Header>
+      </Table.Cell>
+      <Table.Cell header>
+        <Header
+          sortDirection={props.state?.field === ApcTableHeaderColumns.CellState ? props.state.dir : null}
+          onSortClick={() => props.setSortBy(ApcTableHeaderColumns.CellState)}>
+          Cell State
+        </Header>
       </Table.Cell>
-      <Table.Cell header>Cell State</Table.Cell>
     </>
   );
 };
 
+const apcDataComparer
+  = (a: PowerMonitorApcItemData,
+    b: PowerMonitorApcItemData,
+    names: Record<string, string>,
+    field: ApcTableHeaderColumns): number => {
+
+    if (field === ApcTableHeaderColumns.Area) {
+      return names[a[field]].localeCompare(names[b[field]]);
+    } else {
+      return numericCompare(a[field], (b[field]));
+    }
+  };
+
+const sortApcData
+  = (data: PowerMonitorApcItemData[],
+    names: Record<string, string>,
+    state: ApcTableHeaderColumnSortState): PowerMonitorApcItemData[] => {
+
+    if (state !== null) {
+      data.sort((a, b) => apcDataComparer(a, b, names, state.field));
+      if (state.dir === SortDirection.Asc) {
+        data.reverse();
```

Comment:
Also mutates. That's fine if you're working on `sortedData` as above (and then return it below).